numpyro/distributions/continuous.py

```diff
@@ -40,7 +40,7 @@ class cauchy_gen(jax_continuous):
     def _rvs(self):
         # TODO: move this implementation upstream to jax.random.standard_cauchy
         # Another way is to generate X, Y ~ Normal(0, 1) and return X / Y
-        u = random.uniform(self._random_state, self._size)
+        u = random.uniform(self._random_state, shape=self._size)
```

Comment:
This seems like a fine change and isn't blocking this PR, but it will be good to understand why this leads to recompilation.

Reply:
I am curious too so I am investigating it. This is True at least for gamma sampler.